struct.go

```diff
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
+	"github.com/nu7hatch/gouuid"
```

Comment:
As of now, it is called `Enlist` to match the name of the function in q that creates a list. I can rename it if you want me to

Reply:
Oh I see, but Generic takes an interface, will we be able to just replace the Generic call with Enlist?

Reply:
Actually we can't have a Generic function because the type 0 only represents a generic list and not a generic object. We have to use the `q.Marchal` method to convert the interface to `K` object and then send it to query